contracts/src/deployers/ezeth/EzETHHyperdriveDeployerCoordinator.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+import { IERC20 } from "../../interfaces/IERC20.sol";
+import { IHyperdrive } from "../../interfaces/IHyperdrive.sol";
+import { IHyperdriveDeployerCoordinator } from "../../interfaces/IHyperdriveDeployerCoordinator.sol";
+import { IRestakeManager } from "../../interfaces/IRestakeManager.sol";
+import { FixedPointMath, ONE } from "../../libraries/FixedPointMath.sol";
+import { HyperdriveDeployerCoordinator } from "../HyperdriveDeployerCoordinator.sol";
+
+/// @author DELV
+/// @title EzETHHyperdriveDeployerCoordinator
+/// @custom:disclaimer The language used in this code is for coding convenience
+/// @notice The deployer coordinator for the EzETHHyperdrive implementation.
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+contract EzETHHyperdriveDeployerCoordinator is HyperdriveDeployerCoordinator {
+    using FixedPointMath for uint256;
+
+    /// @notice The Renzo contract.
+    IRestakeManager public immutable restakeManager;
+    /// @notice The ezETH token contract.
+    IERC20 public immutable ezETH;
+
+    /// @notice Instantiates the deployer coordinator.
+    /// @param _coreDeployer The core deployer.
+    /// @param _target0Deployer The target0 deployer.
+    /// @param _target1Deployer The target1 deployer.
+    /// @param _target2Deployer The target2 deployer.
+    /// @param _target3Deployer The target3 deployer.
+    /// @param _target4Deployer The target4 deployer.
+    /// @param _restakeManager The Renzo contract.
+    constructor(
+        address _coreDeployer,
+        address _target0Deployer,
+        address _target1Deployer,
+        address _target2Deployer,
+        address _target3Deployer,
+        address _target4Deployer,
+        IRestakeManager _restakeManager
+    )
+        HyperdriveDeployerCoordinator(
+            _coreDeployer,
+            _target0Deployer,
+            _target1Deployer,
+            _target2Deployer,
+            _target3Deployer,
+            _target4Deployer
+        )
+    {
+        restakeManager = _restakeManager;
+        (, bytes memory data) = address(restakeManager).call(
+            abi.encodeWithSignature("ezETH()")
+        );
+        ezETH = IERC20(abi.decode(data, (address)));
```

Comment:
Why isn't this just in the interface? It seems like a view function too, so if we were going to do a low-level call, a staticcall would be better.